### DISCUSSION TITLE
lldpd

### DISCUSSION THREAD

**Discussion by thiamata** [MichaIng/DietPi#6966] [Q&A]
Hi

I am doing some deeper tests with k3s and dietpi also with an automated pre setup process for the nodes.

So I think it could be (for troubleshooting it is)  very helpful to identify which node is connected to which switch port, to correlate the switch port settings.

Therefor I am using the protocol lldp. An older but useful L2-Identifier, used by switches and routers.

A package is available for Linux, lldpd. (apt install lldpd, thatś all)

But on dietpi this installation is needed to be done manually, or better to say after the automated initial setup process from dietpi.

Getting this integrated in the automated setup process would be nice (for me).

Okay, right you need switches, which knows how to handle lldp package. Not really normal for Home usage, but for business stuff it is.

regards Thiamata

**Comment by Joulinar** [comment]
You can execute `Custom Scripts` as part of the initial setup. This way you can install your packages automatically. 

https://github.com/MichaIng/DietPi/blob/master/dietpi.txt#L67-L72
